src/tools/crawler/main.rs

```diff
@@ -77,7 +84,14 @@ async fn main() {
     let args = Args::parse();
 
     // Create the crawler with the given listener address.
-    let crawler = Crawler::new().await;
+    let mut crawler = Crawler::new().await;
+
+    if let Some(peer_file) = args.peer_file {
+        if let Ok(js) = fs::read_to_string(peer_file) {
+            crawler.peer_list = serde_json::from_str(&js).expect("failed to parse peer file");
+            info!(parent: crawler.node().span(), "loaded peer list for {:?} nodes", crawler.peer_list.len());
+        }
+    }
```

Comment:
so the crawler will read the file only once and always suggest the same peer list throughout its runtime? 🕵️‍♀️

Reply:
I think so.
The main flow is that crawler works and creates sample file for crunchy. So the crawler need to finish its job. Crunchy is taking that file and (if the network is big) after some hours produces peer proposal to be used as next crawler run. 
And that how it goes...

I can't see right now any way to increase dynamics as the more crawler is running the more nodes it discovers. Discovering nodes changes the network and graphs and therefore, it could change the ways crunchy and IPS are working. 

I think that it would be safe to work over some completed state - so crunchy works over completed network crawl and crawler works having last stable peerlist.

Reply:
hmhm, maybe we can change so that the peer lists here... are suggested only once - like consumed!

maybe, we could have an automatic cycle like this:
```
1. run crawler
 <- no input
 -> continues running for N amount of time
 -> outputs the network state and dies

loop {
    2. run crunchy
     <- consumes the latest current state
     -> outputs IPS the algo peer lists
     
    3. run crawler
     <- applies (consumes) the latest peer lists
     -  continues running for N amount of time
     -> outputs the network state and dies
}
```

Reply:
> hmhm, maybe we can change so that the peer lists here... are suggested only once - like consumed!

or we just make it so it periodically checks for some fresh IPS suggestion file, once it finds it - it applies and then waits for a new file to be generated by crunchy?